### PR TITLE
Added GoDaddy provider

### LINKED ISSUE #10: Add GoDaddy support
Add GoDaddy support.

See API Documentation:

[https://developer.godaddy.com/doc/endpoint/domains#/v1/recordGet](https://developer.godaddy.com/doc/endpoint/domains#/v1/recordGet)

### PR DESCRIPTION
**Pull request by TheDanielDoyle**
* Added GoDaddy provider.
* Modified Hydrator to be a Mutator and added Mutations to abstract out performing changes to DNSRecordCollection.

### CODE REVIEW

File: src/DDNSUpdate.Tests/Application/DNSRecordCollectionMutatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+using System.Linq;
+using DDNSUpdate.Application;
+using DDNSUpdate.Application.Mutations;
+using DDNSUpdate.Domain;
+using DDNSUpdate.Tests.Helpers;
+using Xunit;
+
+namespace DDNSUpdate.Tests.Application
+{
+    public class DNSRecordCollectionMutatorTests : TestBase
```
Comment (mickey-willis) [inline src/DDNSUpdate.Tests/Application/DNSRecordCollectionMutatorTests.cs]: Does this need to extend the TestBase?

File: src/DDNSUpdate/Application/Providers/GoDaddy/Configuration/GoDaddyDNSRecordValidator.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        public static readonly string NameErrorMessage = "{PropertyName} must be set.";
+        public static readonly string PortErrorMessage = _propertyDisallowedMessage;
+        public static readonly string PriorityErrorMessage = _propertyDisallowedMessage;
+        public static readonly string TagErrorMessage = _propertyDisallowedMessage;
+        public static readonly string TTLErrorMessage = $"{{PropertyName}} must be {_ttlMinimumSeconds} seconds or greater.";
+        public static readonly string TypeErrorMessage = "Use either A or AAA for the {PropertyName}.";
+        public static readonly string WeightErrorMessage = _propertyDisallowedMessage;
+
+        public GoDaddyDNSRecordValidator()
+        {
```
Comment (mickey-willis) [inline src/DDNSUpdate/Application/Providers/GoDaddy/Configuration/GoDaddyDNSRecordValidator.cs:23]: Should there be a check for the ID? Should we make sure that this value is not set?

**Review by mickey-willis — DISMISSED** [review]

File: src/DDNSUpdate/Application/Providers/GoDaddy/Configuration/GoDaddyDNSRecordValidator.cs (hunk 2 above)
Comment (TheDanielDoyle) [reply src/DDNSUpdate/Application/Providers/GoDaddy/Configuration/GoDaddyDNSRecordValidator.cs:23]: Done

File: src/DDNSUpdate.Tests/Application/DNSRecordCollectionMutatorTests.cs (hunk 1 above)
Comment (TheDanielDoyle) [reply src/DDNSUpdate.Tests/Application/DNSRecordCollectionMutatorTests.cs]: Done

**Pull request merged**